lib/memory_cache.dart

```diff
@@ -0,0 +1,27 @@
+/// Simple class to cache values on the cache
+///
+class MemoryCache<K, V> {
+  final Map<K, V> _cache = {};
+  final List<K> _addedOrder = [];
```

Comment:
Can't you just use an ordered map instead?

Reply:
Good call, felt like a noob now.

Made the change.